xlators/cluster/afr/src/afr-self-heal-data.c

```diff
@@ -353,6 +353,12 @@ afr_selfheal_data_do(call_frame_t *frame, xlator_t *this, fd_t *fd, int source,
     }
 
     block = 128 * 1024 * priv->data_self_heal_window_size;
+    if (HAS_HOLES((&replies[source].poststat))) {
```

Comment:
For future improvements, self heal could make use of SEEK_HOLE and SEEK_DATA subcommands of seek() to efficiently handle this case.

Reply:
yeah, that is what I meant by the comment. I saw it was done in rebalance, we should do the same in both EC and replicate.